llvm/lib/TableGen/TGParser.cpp

```diff
@@ -330,17 +330,10 @@ bool TGParser::AddSubClass(Record *CurRec, SubClassReference &SubClass) {
 
   // Since everything went well, we can now set the "superclass" list for the
   // current record.
-  for (const auto &[SC, Loc] : SC->getSuperClasses()) {
-    if (CurRec->isSubClassOf(SC))
-      return Error(SubClass.RefRange.Start,
-                   "Already subclass of '" + SC->getName() + "'!\n");
-    CurRec->addSuperClass(SC, Loc);
-  }
-
```

Comment:
This and some other changes are not obvious to me.
Can you confirm that `-print-records` produces char-identical result on e.g. X86.td?

Reply:
Confirmed. I also compared all generated `.inc` files in an LLVM build tree, with no differences.

For this particular hunk, given `def D : C`, previously we had to add all C's superclasses, as well as C itself, to D's superclass list. Now we only have to add C to D's direct superclass list.

Reply:
Thanks!